### PR TITLE
Add edit task and deadline

### LINKED ISSUE #84: feat(tasks): Edit module associated to task
# User Story

Tasks/Deadlines User Story 11

As a user, I can change the module of a specific task or deadline, so that I can move the tasks around if I make a mistake.

### PR DESCRIPTION
**Pull request by HoJunHao2000**
Fix #84 #86

### CODE REVIEW

**Review by jontmy — CHANGES_REQUESTED** [review]
Code looks good to merge, but we should include unit tests for new features this iteration.

File: src/main/java/modtrekt/logic/commands/EditTaskCommand.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                ? ((Deadline) taskToEdit).getDueDate()
+                : null;
+
+        Task newTask = deadline != null
+                ? new Deadline(description, code, deadline, archived)
+                : new Task(description, code, archived);
+
+        model.deleteTask(taskToEdit);
+        model.addTask(newTask);
+        model.updateModuleTask(newTask);
```
Comment (domoberzin) [inline src/main/java/modtrekt/logic/commands/EditTaskCommand.java]: this line is redundant I think, addTask calls this method already

Comment (HoJunHao2000) [reply src/main/java/modtrekt/logic/commands/EditTaskCommand.java]: removed

**Comment by HoJunHao2000** [comment]
> Code looks good to merge, but we should include unit tests for new features this iteration.

added unit tests

File: src/main/java/modtrekt/logic/parser/EditTaskCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        System.out.println(description);
+
+        Optional<LocalDate> optionalDeadline =
+                command.mapOptionalArgument(PREFIX_DEADLINE, rethrowFunction(ParserUtil::parseDueDate));
+        LocalDate deadline = optionalDeadline.isPresent() ? optionalDeadline.get() : null;
+
+        if (deadline == null && code == null && description == null) {
+            throw new ParseException(
+                    "Please enter a deadline(-d) and/or module code(-c) and/or description(-ds) to edit");
+        }
```
Comment (jontmy) [inline src/main/java/modtrekt/logic/parser/EditTaskCommandParser.java:58]: Converting an `Optional` to a nullable value and then testing for `null` can be replaced by `Optional::orElseThrow` - didn't see this in the earlier review, my bad.

**Review by jontmy — CHANGES_REQUESTED** [review]
ignore this, pr wasn't actually ready for review

**Pull request merged**